### DISCUSSION TITLE
Fake symbolic link from APG MOD

### DISCUSSION THREAD

**Discussion by parsec82** [Mr-MIBonk/M.I.B._More-Incredible-Bash#4] [Ideas]
after some test on 2 unit i have see on efs-persist
`/mnt/efs-persist/FEC` is not a directory but a symbolic link that go to `/protect/FEC` (this directory not exist)

![image](https://user-images.githubusercontent.com/5319830/152875509-e93686fd-e15b-4d87-a322-e83b7e5bd2c0.png)

if you use MIB Tool without delete this it go on error.

i suggest to edit the `CreateFec` function on `addfec` file and start the function with the add of a IF statement that do this:

```
if [ FEC is a symbolic link? ]
    delete FEC symbolic link
    create FEC directory
fi
```

i think the script have to do this:

```
if [[ -L "/mnt/efs-persist/FEC" ]]; then
    rm -f /mnt/efs-persist/FEC
    mkdir /mnt/efs-persist/FEC
fi
```

**Comment by harman-f** [comment]
I'm not familiar with '-L' option.
It is checking for simbolic links only?

has to be tested on unit.


FecContainer.fec in /net/rcc/mnt/efs-system/backup/FEC/ is deletetd, not existing?

**Reply by parsec82** [reply]
Yes on my Linux Distro with `-L` you can check if the file is a symbolic link.

on APG unit the `FecContainer.fec` is not present, because `FEC` is not a directory.

Tomorrow i have on my bench an Audi A4 8W unit modded by APG, i can check the `-L` argument and i can create (if needed) a full backup of the unit (except the NavDB partition) and share for check if is present other file... or where is the `FecContainer.fec` on APG mod.

**Comment by harman-f** [comment]
As the unit has to be cleaned manually anyway, it will be easier to just do this as well.
Please add to mibwiki ;)

**Comment by Mr-MIBonk** [comment]
APG Protection is the the biggest shit i have ever seen! Only way is to cleanup the Unit with complete new Installation. Everything else is not possible free of residues and without side effects!
